radicle-node/src/service.rs

```diff
@@ -554,6 +567,38 @@ where
         }
     }
 
+    pub fn ping(&mut self, peer_ip: &net::IpAddr) -> Result<(), peer::SessionError> {
+        let peer = if let Some(peer) = self.sessions.get_mut(peer_ip) {
+            peer
+        } else {
+            return Err(SessionError::NotFound(*peer_ip));
+        };
+
+        let ping_upper = 16;
+        let ponglen = self.rng.u16(1..ping_upper);
+        let msg = Message::Ping {
+            ponglen,
+            zeroes: message::ZeroBytes(self.rng.u16(1..ping_upper)),
+        };
+        self.reactor.write(peer.addr, msg);
+
+        // How can I access the messages' ponglen field?
+        peer.ping = PingState::AwaitingResponse(ponglen);
```

Comment:
```suggestion
        peer.ping = PingState::AwaitingResponse(ponglen);
        
```
Why do you need to access the field?